ui/ducks/send/send.js

```diff
@@ -0,0 +1,1442 @@
+        conversionUtil(tokenAmountInDec, {
+          fromNumericBase: 'dec',
+          toNumericBase: 'hex',
+        }),
+      );
+
+      await dispatch(
+        updateSendAsset({
+          type: ASSET_TYPES.TOKEN,
+          details: { ...assetDetails, address: tokenAddress },
+        }),
+      );
+
+      await dispatch(
+        actions.editTransaction({
+          id: transactionId,
+          gasLimit,
+          gasPrice,
+          from,
+          amount: tokenAmountInHex,
+          address,
+          nickname,
+        }),
+      );
+    }
+  };
+}
+
+// Selectors
+
+// Gas selectors
+export function getGasLimit(state) {
+  return state[name].gas.gasLimit;
+}
+
+export function getGasPrice(state) {
+  return state[name].gas.gasPrice;
+}
+
+export function getGasTotal(state) {
+  return state[name].gas.gasTotal;
+}
+
+export function gasFeeIsInError(state) {
+  return Boolean(state[name].gas.error);
+}
+
+export function getMinimumGasLimitForSend(state) {
+  return state[name].gas.minimumGasLimit;
+}
+
+export function getGasInputMode(state) {
+  const isMainnet = getIsMainnet(state);
+  const showAdvancedGasFields = getAdvancedInlineGasShown(state);
+  if ((!isMainnet && !process.env.IN_TEST) || showAdvancedGasFields) {
+    return GAS_INPUT_MODES.INLINE;
+  }
+  return state[name].gas.mode;
+}
+
+// Asset Selectors
+
+export function getSendAsset(state) {
+  return state[name].asset;
+}
+
+export function getSendAssetAddress(state) {
+  return getSendAsset(state)?.details?.address;
+}
+
+// Amount Selectors
+export function getSendAmount(state) {
+  return state[name].amount.value;
+}
+
+export function getIsBalanceInsufficient(state) {
+  return state[name].gas.error === INSUFFICIENT_FUNDS_ERROR;
+}
+export function getSendMaxModeState(state) {
+  return state[name].amount.mode === AMOUNT_MODES.MAX;
+}
+
+export function getSendHexData(state) {
+  return state[name].draftTransaction.userInputHexData;
+}
+
+export function sendAmountIsInError(state) {
+  return Boolean(state[name].amount.error);
+}
+
+// Recipient Selectors
+
+export function getSendTo(state) {
+  return state[name].recipient.address;
+}
+
+export function getIsUsingMyAccountForRecipientSearch(state) {
+  return state[name].recipient.mode === RECIPIENT_SEARCH_MODES.MY_ACCOUNTS;
+}
+
+export function getRecipientUserInput(state) {
+  return state[name].recipient.userInput;
+}
+
+export function getRecipient(state) {
+  return state[name].recipient;
+}
+
+// Overall validity and stage selectors
+
+export function getSendErrors(state) {
+  return {
+    gas: state.send.gas.error,
+    amount: state.send.amount.error,
+  };
+}
+
+export function isSendStateInitialized(state) {
+  return state[name].stage !== SEND_STAGES.UNINITIALIZED;
+}
+
+export function isSendFormInvalid(state) {
+  return state[name].status === SEND_STATUSES.INVALID;
+}
+
+export function isSendFormInError(state) {
+  return Boolean(state[name].amount.error || state[name].gas.error);
```

Comment:
The old `isSendFormInError` also used to return `true` if gas hadn't finished loading. I see that the "gasLoadingError" is now set to whether the send state is initialized. Perhaps that should be included in this expression as well? Or is submission during initialization already being prevented through some other means?

Reply:
Yeah, the button in the footer to submit is not enabled if the send state is 'INVALID' which it will be during initialization. This selector is now *only* used for gathering data for metrics events.